Genuine Carrot Goat

Medium

# `_swapV3Single()` will revert on Arbitrum due to the catch block trying to get a pool with a fee tier of 0.02% (which is non-existent in UniV3)

### Summary

`_swapV3Single()` will revert when called on Arbitrum due to the faulty try/catch block.

First, it will try to get the pool with token0, token1, and 1% pool fee tier, but on Arbitrum currently, only 0.3% fee tier exists as per Peapods website: https://docs.peapods.finance/links/contract-addresses

Second of all, the revert will be caught gracefully in the catch block, but the issue is that in the catch block, it tries to get the pool with a 0.02% fee tier, which is non-existent in Uniswap v3, causing a total revert of the function.

### Root Cause

I would consider the root cause to be the catch block, as it should NOT contain code that can revert, so the root cause is trying to retrieve a pool with a fee tier which does not exist on uni v3, which is the 0.02% pool fee tier.

But we can't leave the part in which the try block tries to retrieve the pool with a fee tier which exists in uni v3 -> 1%, but there are no currently pools deployed on Arbitrum with 1% fee tier as seen in Peapods' website.

### Internal Pre-conditions

The blockchain needs to be Arbitrum.

### External Pre-conditions

None

### Attack Path

1. `indexUtils::addLPAndStake()` is called to add some tokens into the LP pool, which happens during leveraging
2. The transaction will revert due to it going into `_zap()` and into `_swapV3Single()` which first reverts because no pool with 1% fee exists on Arbitrum, but this will be caught in the catch block, but then the catch block itself tries to retrieve a pool with 0.02% fee tier, which does not exists as a possibility in Uni v3

### Impact

Leverage can't happen on Arbitrum and will revert

### PoC

_No response_

### Mitigation

The catch block should try to retrieve the pool with an existing and deployed fee tier.